community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -103,19 +101,16 @@
  * </pre>
 
  * The writes that happened before the last checkpoint are durable and safe, but the writes after it are not.
- * The tree can however get back to a consistent state by:
- * <ol>
- * <li>Creator of this tree detects that recovery is required (i.e. non-clean shutdown) and if so must call
- * {@link #bumpUnstableGeneration()} ones, before any writes during recovery are made.</li>
- * <li>Replaying all the writes, exactly as they were made, since the last checkpoint all the way up
- * to the crash ({@code x}). Even including writes before the last checkpoint is OK, important is that
- * <strong>at least</strong> writes since last checkpoint are included.
- * </ol>
+ * The tree can however get back to a consistent state by replaying all the writes, exactly as they were made,
```

Comment:
Good update to the javadocs. `exactly as they were made` sounds stricter than it actually is and somewhat vague at the same time. It could instead mention that all changes has to be replayed (which is well described in your changes below), but that the order isn't strictly that of the changes as they happened in the first place.